internal/provider/data_source_instance.go

```diff
@@ -0,0 +1,103 @@
+package provider
+
+import (
+	webclient "axual-webclient"
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ datasource.DataSource = &instanceDataSource{}
+
+func NewInstanceDataSource(provider AxualProvider) datasource.DataSource {
+	return &instanceDataSource{
+		provider: provider,
+	}
+}
+
+type instanceDataSource struct {
+	provider AxualProvider
+}
+
+type instanceDataSourceData struct {
+	Name        types.String `tfsdk:"name"`
+	ShortName   types.String `tfsdk:"short_name"`
+	Description types.String `tfsdk:"description"`
+	Id          types.String `tfsdk:"id"`
+}
+
+func (d *instanceDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_instance"
+}
+
+func (d *instanceDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Instance resource. Read more: https://docs.axual.io/axual/2024.2/self-service/instance-management.html",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Instance's unique identifier",
+				Computed:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Instance's name",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(3, 80),
+					stringvalidator.RegexMatches(regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9 ._-]*$`), "can only contain letters, numbers, dots, spaces, dashes and underscores, but cannot begin with an underscore, dot, space or dash"),
```

Comment:
The API Java regex allows the string to start with a separator (dot, dash, underscore, or space).
The Terraform Go regex requires the string to start with an alphanumeric character (A-Za-z0-9).
Please change Terraform regex to match API Java regex.